core-tests/shared/src/test/scala/zio/stm/ZSTMSpec.scala

```diff
@@ -1239,7 +1239,27 @@ object ZSTMSpec extends ZIOBaseSpec {
         _     <- transaction(ref).commit
         value <- ref.get
       } yield assertTrue(value == 9)
-    } @@ exceptJS(nonFlaky(10000))
+    } @@ exceptJS(nonFlaky(10000)),
+    test("doesn't deadlock under concurrency when transactions create new ZSTMs - i9215") {
+      def transaction(arr: TArray[String]) =
+        ZSTM.suspend {
+          Thread.sleep(1) // Force ZSTM lock contention
```

Comment:
Looks like it fails with Scala.js despite the except aspect

Reply:
Aghh the except part is for `nonFlaky`. Seems that ScalaJS doesn't like the `Thread.sleep` part, I'll move the test to be run for JVM/Native only